mergify_engine/tests/unit/rules/test_filter.py

```diff
@@ -154,6 +157,242 @@ async def test_str() -> None:
+    assert await f(FakePR({"foo": time(5, 8)}))
+    assert not await f(FakePR({"foo": time(2, 1)}))
+    assert not await f(FakePR({"foo": time(5, 1)}))
+    assert await f(FakePR({"foo": time(6, 2)}))
+    assert await f(FakePR({"foo": time(8, 9)}))
+
+
+@freeze_time("2012-01-14T12:15:00")
+async def test_time_near_datetime() -> None:
+    atnoon = datetime.datetime(2012, 1, 15, 0, 0, 0, 0, tzinfo=datetime.timezone.utc)
+    nextday = datetime.datetime(2012, 1, 15, 5, 8, 0, 0, tzinfo=datetime.timezone.utc)
+    today = datetime.datetime(2012, 1, 14, 5, 8, 0, 0, tzinfo=datetime.timezone.utc)
+    soon = datetime.datetime(2012, 1, 14, 5, 9, 0, 0, tzinfo=datetime.timezone.utc)
+
+    f = filter.NearDatetimeFilter({"<=": ("foo", today.timetz())})
+    assert await f(FakePR({"foo": time(5, 8)})) == soon
+    assert await f(FakePR({"foo": time(2, 1)})) == today
+    assert await f(FakePR({"foo": time(5, 1)})) == today
+    assert await f(FakePR({"foo": time(6, 2)})) == atnoon
+    assert await f(FakePR({"foo": time(8, 9)})) == atnoon
+
+    f = filter.NearDatetimeFilter({"<": ("foo", today.timetz())})
+    assert await f(FakePR({"foo": time(5, 8)})) == atnoon
+    assert await f(FakePR({"foo": time(2, 1)})) == today
+    assert await f(FakePR({"foo": time(5, 1)})) == today
+    assert await f(FakePR({"foo": time(6, 2)})) == atnoon
+    assert await f(FakePR({"foo": time(8, 9)})) == atnoon
+
+    f = filter.NearDatetimeFilter({">=": ("foo", today.timetz())})
+    assert await f(FakePR({"foo": time(5, 8)})) == soon
+    assert await f(FakePR({"foo": time(2, 1)})) == today
+    assert await f(FakePR({"foo": time(5, 1)})) == today
+    assert await f(FakePR({"foo": time(6, 2)})) == atnoon
+    assert await f(FakePR({"foo": time(8, 9)})) == atnoon
+
+    f = filter.NearDatetimeFilter({">": ("foo", today.timetz())})
+    assert await f(FakePR({"foo": time(5, 8)})) == atnoon
+    assert await f(FakePR({"foo": time(2, 1)})) == today
+    assert await f(FakePR({"foo": time(5, 1)})) == today
+    assert await f(FakePR({"foo": time(6, 2)})) == atnoon
+    assert await f(FakePR({"foo": time(8, 9)})) == atnoon
+
+    f = filter.NearDatetimeFilter({"=": ("foo", today.timetz())})
+    assert await f(FakePR({"foo": time(5, 8)})) == soon
+    assert await f(FakePR({"foo": time(2, 1)})) == today
+    assert await f(FakePR({"foo": time(5, 1)})) == today
+    assert await f(FakePR({"foo": time(6, 2)})) == nextday
+    assert await f(FakePR({"foo": time(8, 9)})) == nextday
+
+    f = filter.NearDatetimeFilter({"!=": ("foo", today.timetz())})
+    assert await f(FakePR({"foo": time(5, 8)})) == soon
+    assert await f(FakePR({"foo": time(2, 1)})) == today
+    assert await f(FakePR({"foo": time(5, 1)})) == today
+    assert await f(FakePR({"foo": time(6, 2)})) == nextday
+    assert await f(FakePR({"foo": time(8, 9)})) == nextday
+
+    f = filter.NearDatetimeFilter({"-": {"<=": ("foo", today.timetz())}})
+    assert await f(FakePR({"foo": time(5, 8)})) == soon
```

Comment:
Not sure I understand here. If you say `-current-time<=05:08` then it means `current-time>05:08` which means you should schedule at midnight. The result here should be exactly like the L246 `f = filter.NearDatetimeFilter({">": ("foo", today.timetz())})`.